WalletWasabi/WabiSabi/Client/Batching/PaymentBatch.cs

```diff
@@ -21,20 +23,45 @@ public class PaymentBatch
 {
 	private readonly List<Payment> _payments = new();
 	private readonly object _syncObj = new();
-	private IEnumerable<PendingPayment> PendingPayments => GetPayments().OfType<PendingPayment>();
-	private IEnumerable<InProgressPayment> InProgressPayments => GetPayments().OfType<InProgressPayment>();
+	private IEnumerable<Payment> PendingPayments => GetPayments().Where(p => p.State is PendingPayment);
+	private IEnumerable<Payment> InProgressPayments => GetPayments().Where(p => p.State is InProgressPayment);
 
 	public Guid AddPayment(IDestination destination, Money amount)
 	{
-		var payment = new PendingPayment(destination, amount);
+		var payment = new Payment(destination, amount);
 		lock (_syncObj)
 		{
 			_payments.Add(payment);
 		}
-		Logger.LogInfo($"Payment for {payment.Amount} BTC to {payment.Destination.ScriptPubKey}.");
+		Logger.LogInfo($"Payment {payment.Id} for {payment.Amount} BTC to {payment.Destination.ScriptPubKey}.");
 		return payment.Id;
 	}
 
+	public void AbortPayment(Guid id)
+	{
+		lock (_syncObj)
+		{
+			if (_payments.FirstOrDefault(p => p.Id == id) is {} payment)
+			{
+				if (payment.State is PendingPayment)
+				{
+					_payments.Remove(payment);
```

Comment:
I wonder how this works.

I did a `payincoinjoin` and after that I cancelled it. 
It was successfully removed from this list, but the CJ itself will be confirmed once I mine a block. So it's not really cancelled unless the client replaces the CJ with an RBF tx.

So, cancelling payments means only removing from the `_payments` list?